src/Nowruz/modules/settings/components/notification/useNotification.tsx

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useState } from 'react';
+import { NotificationSetting, NotificationType, notificationSettings, updateNotificationSettings } from 'src/core/api';
+import translate from 'src/translations';
+
+import { SettingsItem, ToggleButtonItem } from './payload.type';
+
+export const useNotification = () => {
+  const [settings, setSettings] = useState<NotificationSetting[]>([]);
+  const [mappedSettings, setmappedSettings] = useState<SettingsItem[]>([]);
+  const [allChecked, setAllChecked] = useState(false);
+
+  useEffect(() => {
+    notificationSettings().then((e) => {
+      setSettings(e.settings);
+    });
+  }, []);
+
+  const changeSetting = async (type: string, toggleButtonLabel: 'email' | 'in_app' | 'push') => {
+    const settingVal = [...settings];
+    const idx = settingVal.findIndex((item) => item.type === type);
+    settingVal[idx][toggleButtonLabel] = !settingVal[idx][toggleButtonLabel];
+    setSettings(settingVal);
+  };
+
+  const getToggleButtons = (type: NotificationType) => {
+    return [
+      {
+        text: 'Push',
+        checked: settings.find((s) => s.type === type)?.push || false,
+        onChange: () => changeSetting(type, 'push'),
+      },
+      {
+        text: 'In-app',
+        checked: settings.find((s) => s.type === type)?.in_app || false,
+        onChange: () => changeSetting(type, 'in_app'),
+      },
+      {
+        text: 'Email',
+        checked: settings.find((s) => s.type === type)?.email || false,
+        onChange: () => changeSetting(type, 'email'),
+      },
+    ];
+  };
+
+  const mapSetting = () => {
+    const isAllChecked = settings.every((s) => s.in_app && s.email && s.push);
+    setAllChecked(isAllChecked);
+    let mapped: SettingsItem[] = [];
+
+    let label = 'Allow notifications';
+    let description = '';
+    let list: ToggleButtonItem[] = [
+      {
+        text: 'Allow all',
+        checked: isAllChecked,
+        onChange: onAllowNotifications,
+      },
+    ];
+
+    mapped.push({ label, description, toggleButtons: list });
```

Comment:
please create arrays of objects and put into `mapped` once.
```
mapped = [
{ label: 'Allow notifications', description: '', toggleButtons: [{...}] },
{ label: 'Likes', ... },
{ label: 'Comments', ... }
]

Reply:
Let's skip this, the design has changed and we should change this page totally later